### ISSUE TITLE
D1 API dialect

### ISSUE THREAD

**Issue by tom-sherman** [aidenwallis/kysely-d1#7]
A D1 API dialect would be cool, it would allow you use kysely in node or deno to work with D1 databases

**Comment by tom-sherman** [comment]
I've created a POC implementation here: https://github.com/tom-sherman/kysely-d1-api

Let me know if you think this is something that could be included in this lib and I'll be happy to clean it up and open a pull request. If not no worries, I can publish it as a complimentary library.

**Comment by aidenwallis** [comment]
I think we can merge this in, can you open a PR and I can check it out then?

**Comment by aidenwallis** [comment]
Apologies for completely missing the second part of that, if the API is not formally supported by Cloudflare, I don't want to merge it in here and encourage use of APIs that they may not want people using. I'm happy to revisit this when/if they document the API for this!

**Comment by tom-sherman** [comment]
Cloudflare are happy for users to use this API, they even promoted a tool using it in the open alpha blog post. Totally get the point about only shopping stable APIs in this library though 👍

**Comment by aidenwallis** [comment]
Yeah, I think we should totally work on adding this FWIW and would like to combine it into this library!

I just don't want to risk shipping a library that breaks for people randomly, lets keep this open and revisit merging the two projects once the API is documented?

**Comment by ngraef** [comment]
Any interest in picking this back up now that the [D1 REST API](https://developers.cloudflare.com/api/operations/cloudflare-d1-query-database) is public, stable, and documented?

**Comment by aidenwallis** [comment]
If someone picks up a PR or submits another, I'm happy to take a look and hopefully merge. To be a little transparent, i don't really use this package at all anymore and have been extremely busy with work, but will try to look at this when I have time. Thank you for bearing with me and sorry for the delays. 👍
